packages/core-v2/src/visual-model/visual-entity.ts

```diff
@@ -1,16 +1,133 @@
-export type Position = {
+import { Entity, EntityIdentifier } from "./entity-model/entity";
+
+/**
+ * Color in hexadecimal, must start with "#" character.
+ * Should be in lower case.
+ */
+export type Color = string;
+
+/**
+ * Base interface for all visual entities.
+ */
+export interface VisualEntity extends Entity { }
+
+/**
+ * Represents an entity, i.g. class or a profile.
+ */
+export interface VisualNode extends VisualEntity {
+
+    /**
+     * Identifier of represented entity.
+     */
+    representedEntity: EntityIdentifier;
+
+    /**
+     * Identifier of the entity model.
+     */
+    model: string;
+
+    /**
+     * Position on canvas.
+     */
+    position: Position;
+
+    /**
+     * Identifiers of non-visual relationships to show as a part of the entity.
+     */
+    content: string[];
+
+    /**
+     * List of visual models assigned to this entity as an internal representation.
+     */
+    visualModels: string[];
```

Comment:
I am not sure what this means. I have 2 interpretations:

1. It is all the diagrams where this specific visual entity is visible (i. e. it is in the concrete visual model). But I think that currently one concerete semantic class is represented in different visual models by different visual entities.

2. It is the linked visual models we talked about. So some visual models, which are assigned to this visual entity as representative diagrams.

Reply:
Hopefully the new comment explains it better.